pkg/apis/core/types.go

```diff
@@ -2593,6 +2593,13 @@ type PodSpec struct {
 	// configuration based on DNSPolicy.
 	// +optional
 	DNSConfig *PodDNSConfig
+	// Share a single process namespace between all of the containers in a pod.
+	// When this is set containers will be able to view and signal processes from other containers
+	// in the same pod, and the first process in each container will not be assigned PID 1.
+	// HostPID and ShareProcessNamespace cannot both be set.
+	// Optional: Default to false.
+	// +optional
+	ShareProcessNamespace *bool
```

Comment:
What's the argument for or against putting this into `PodSecurityContext` (alongside HostPID)?

Reply:
If there's an argument against, should we be creating a new nested struct that contains process or namespace settings?

How much stuff should be at the top level of PodSpec, vs more properly isolated?

Reply:
If we put it in PodSecurityContext, is there a reason to call this `SharedPID` instead of `SharedProcessNamespace`?  Or should it be `ProcessNamespace string` with constant values `Host`, `Shared`, `...` for future extension?

Reply:
I think putting it in `PodSecurityContext` is reasonable, but I'm lacking context on its intended use so I wanted to discuss it with you prior to writing custom conversion to the flattened `v1.PodSpec`. In my mind, this isn't really a security setting, but based on community response to this feature I think many disagree.

I started with `SharedPID` as the name, but now I agree with your [initial assessment](https://github.com/kubernetes/community/pull/1048#discussion_r159146536) that it isn't worth keeping a nondescriptive name just for the sake of consistency.

An enum is the [approach we took in the CRI](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/pod-pid-namespace.md#container-runtime-interface-changes), but that API is still in alpha. It actually didn't occur to me to do something similar in the core API but keep the bools in `v1`, is that what you mean? That seems reasonable to me, but I haven't *fully* grokked how the versioned/nonversioned interfaces are used.